core/src/main/java/tc/oc/pgm/command/ListCommand.java

```diff
@@ -5,96 +5,106 @@
 import java.util.List;
 import java.util.UUID;
 import java.util.stream.Collectors;
-import net.md_5.bungee.api.ChatColor;
+import net.kyori.text.Component;
+import net.kyori.text.TextComponent;
+import net.kyori.text.TranslatableComponent;
+import net.kyori.text.format.TextColor;
 import org.bukkit.command.CommandSender;
 import tc.oc.pgm.api.PGM;
 import tc.oc.pgm.api.Permissions;
 import tc.oc.pgm.api.match.Match;
 import tc.oc.pgm.api.player.MatchPlayer;
 import tc.oc.pgm.teams.TeamMatchModule;
-import tc.oc.pgm.util.component.Component;
-import tc.oc.pgm.util.component.Components;
-import tc.oc.pgm.util.component.types.PersonalizedText;
-import tc.oc.pgm.util.component.types.PersonalizedTranslatable;
+import tc.oc.pgm.util.chat.Audience;
 import tc.oc.pgm.util.named.NameStyle;
+import tc.oc.pgm.util.text.TextFormatter;
 
 // TODO: clean up format and use new components
 public final class ListCommand {
 
   @Command(
       aliases = {"list", "who", "online", "ls"},
       desc = "View a list of online players")
-  public void list(CommandSender sender, Match match) {
+  public void list(CommandSender sender, Audience viewer, Match match) {
     TeamMatchModule tmm = match.getModule(TeamMatchModule.class);
     if (tmm != null) {
-      sender.sendMessage(
-          new PersonalizedTranslatable("command.list.teams")
-              .getPersonalizedText()
-              .color(ChatColor.GRAY));
+      viewer.sendMessage(TranslatableComponent.of("command.list.teams", TextColor.GRAY));
       tmm.getTeams()
           .forEach(
               team ->
                   sendTeamInfo(
-                      sender, team.getComponentName(), team.getPlayers(), team.getMaxPlayers()));
+                      viewer,
+                      sender,
+                      team.getComponentName(),
+                      team.getPlayers(),
+                      team.getMaxPlayers()));
     } else {
       sendTeamInfo(
+          viewer,
           sender,
-          new PersonalizedTranslatable("command.list.participants"),
+          TranslatableComponent.of("command.list.participants"),
           match.getParticipants(),
           match.getMaxPlayers());
     }
     // Observers
     sendTeamInfo(
+        viewer,
         sender,
         match.getDefaultParty().getComponentName(),
         match.getDefaultParty().getPlayers(),
         -1);
 
     // Total count
-    sender.sendMessage(
-        new PersonalizedTranslatable(
-                "command.list.online",
-                new PersonalizedText(
-                    Integer.toString(getSize(match.getPlayers(), false)), ChatColor.GREEN))
-            .getPersonalizedText()
-            .color(ChatColor.GRAY));
+    viewer.sendMessage(
+        TranslatableComponent.of("command.list.online", TextColor.GRAY)
+            .args(
```

Comment:
Any `args` calls are unnescary, you can inline them all inside the `of(...)`

Reply:
Oh good to know. I’ll convert the `args` to inline, thanks!